Add DD_AGENT_IPC_* env vars

mackjmr opened the pull request:
#### What this PR does / why we need it:
This PR adds DD_AGENT_IPC_PORT and DD_AGENT_IPC_CONFIG_REFRESH_INTERVAL env vars in otel agent and core agent. It does not override them if they are added by the user. This is necessary for the otel-agent to pull the api key from core config in the case where backend secrets are used.

This PR also adds flag `--sync-delay=30s` to otel agent.

#### Which issue this PR fixes
OTEL-2323

#### Special notes for your reviewer:

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [ ] Chart Version bumped
- [ ] Documentation has been updated with helm-docs (run: `.github/helm-docs.sh`)
- [ ] `CHANGELOG.md` has been updated
- [ ] Variables are documented in the `README.md`
- [ ] For Datadog Operator chart or value changes update the test baselines (run: `make update-test-baselines`)

levan-m reviewed (commented):
Could you please consider adding test like [this](https://github.com/DataDog/helm-charts/blob/main/test/datadog/process_agent_test.go) to make sure chart doesn't break and it works as intended.

mackjmr commented on charts/datadog/templates/_container-agent.yaml:
```diff
@@ -59,6 +59,38 @@
     {{- include "processes-common-envs" . | nindent 4 }}
+    {{- if eq (include "should-enable-otel-agent" .) "true" }}
+    {{- $found := false }}
+    {{- range .Values.agents.containers.agent.env }}
```
Moved logic to helpers in: https://github.com/DataDog/helm-charts/pull/1661/commits/267473ac9b7a1e7e33ff364ef15e28fb71b41c6a.

mackjmr commented on charts/datadog/templates/_container-agent.yaml:
```diff
@@ -59,6 +59,38 @@
+    {{- end }}
+    {{- $found := false }}
+    {{- range .Values.agents.containers.agent.env }}
+    {{- if eq .name "DD_AGENT_IPC_CONFIG_REFRESH_INTERVAL" }}
```
Youre right, we dont set this in helm charts. This is to make sure we don't override it if the user sets it in `datadog.env`, `Values.agents.containers.agent.env` or `Values.agents.containers.otelAgent.env`.

mackjmr commented:
Thanks for the review @levan-m, I've addressed the feedback and added a test in: https://github.com/DataDog/helm-charts/pull/1661/commits/a20b97c473482fd4a2dd7b5bcb8eeb66380298b0. Unfortunately, it seems like I am unable to test the option to not override user provided env vars, as setting something like:
```
					"datadog.env": `
- name: DD_AGENT_IPC_PORT
  value: 1234
- name: DD_AGENT_IPC_CONFIG_REFRESH_INTERVAL
  value: 13`,
```
in the test cases does not reflect these env vars in the test results, so I am guessing this gets done at another stage ? 

I was getting the same issue in not being able to test that we don't override in the operator PR: https://github.com/DataDog/datadog-operator/pull/1604#discussion_r1905453677.

clamoriniere commented on charts/datadog/templates/_container-otel-agent.yaml:
```diff
@@ -3,10 +3,10 @@
   {{- end -}}
   {{- if eq .Values.targetSystem "windows" }}
-  command: ["otel-agent", "-foreground", "-config={{ template "datadog.otelconfPath" . }}/otel-config.yaml", "--core-config={{ template "datadog.confPath" .  }}/datadog.yaml"]
+  command: ["otel-agent", "-foreground", "-config={{ template "datadog.otelconfPath" . }}/otel-config.yaml", "--core-config={{ template "datadog.confPath" .  }}/datadog.yaml", "--sync-delay=30s"]
```
does this new parameter has been always available in the `otel-agent` ? 

If not, what happen if the `otel-agent` version doesn't support this parameter? is it only a warning or an error?

mackjmr replied:
Appears like this was added x.58: https://github.com/DataDog/datadog-agent/pull/28319. Although please note that the otel-agent is still in beta and the only public docs we have: https://docs.datadoghq.com/opentelemetry/agent/install_agent_with_collector/#configure-the-datadog-agent mention x.61.

clamoriniere commented on charts/datadog/templates/_container-agent.yaml:
```diff
@@ -59,6 +59,16 @@
     {{- include "fips-envvar" . | nindent 4 }}
     {{- include "processes-common-envs" . | nindent 4 }}
+    {{- if eq (include "should-enable-otel-agent" .) "true" }}
+    {{- if eq (include "agentIpcPort-envvar-agent-present" .) "false" }}
```
usually we don't need to check if the user set an envvar because we add additional user envvars at the end of the list and so they override other envvars with the same key name.

mackjmr replied:
Good to know, this simplifies the PR a lot. Addressed in https://github.com/DataDog/helm-charts/pull/1661/commits/72eb8feeb23014f29adf9f7d34e13885af5fb104

levan-m reviewed (approved):
Looks good to me. Thanks for adding test!
Could you please update from main, bump chart version and regenerate docs.
